core/src/main/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceFactory.java

```diff
@@ -509,23 +514,23 @@ private static int order(Method method) {
      * Returns a specified path pattern. The path pattern might be specified by {@link Path} or
      * HTTP method annotations such as {@link Get} and {@link Post}.
      */
-    private static String findPattern(Method method, Set<Annotation> methodAnnotations) {
-        String pattern = findFirst(method, Path.class).map(Path::value)
-                                                      .orElse(null);
+    private static Set<String> findPatterns(Method method, Set<Annotation> methodAnnotations) {
+        Set<String> patterns = findAll(method, Path.class).stream().map(Path::value)
+                                                          .collect(Collectors.toSet());
         for (Annotation a : methodAnnotations) {
             final String p = (String) invokeValueMethod(a);
             if (DefaultValues.isUnspecified(p)) {
                 continue;
             }
-            checkArgument(pattern == null,
-                          "Only one path can be specified. (" + pattern + ", " + p + ')');
-            pattern = p;
+            checkArgument(patterns.isEmpty(),
+                          "Only one path can be specified. (" + patterns + ", " + p + ')');
```

Comment:
Just think if error message is hard to understand. it's more like `You can specify path through `Path` or  Http method annotation?